src/main.c

```diff
@@ -508,11 +541,25 @@ int main(int argc, char* argv[]) {
       goto out;
     }
     jq_set_attr(jq, jv_string("PROGRAM_ORIGIN"), jq_realpath(jv_string(dirname(program_origin))));
+    if (jv_object_has(jv_copy(program_arguments), jv_string("ARGS"))) {
+      jv_free(ARGS);
+    } else {
+      ARGS = JV_OBJECT(jv_string("positional"), ARGS,
+                       jv_string("named"), jv_copy(program_arguments));
+      program_arguments = jv_object_set(program_arguments, jv_string("ARGS"), jv_copy(ARGS));
```

Comment:
I think you just put `program_arguments` inside itself as `program_arguments.ARGS.named`...
I'm not sure why, either?